### PR TITLE
	Time different blocks of travis script

### PR DESCRIPTION
**Pull request by phadej**
Makes easier to reason how to address https://github.com/haskell/cabal/issues/3465

### CODE REVIEW

**Comment by ezyang** [comment]
Great; I would champion a PR for this.

On May 30, 2016 8:45:59 AM GMT-05:00, Oleg Grenrus <notifications@github.com> wrote:
>
>You can view, comment on, or merge this pull request online at:
>
>  https://github.com/haskell/cabal/pull/3469
>
>-- Commit Summary --
>
>  * WIP
>
>-- File Changes --
>
>    M travis-script.sh (83)
>
>-- Patch Links --
>
>https://github.com/haskell/cabal/pull/3469.patch
>https://github.com/haskell/cabal/pull/3469.diff
>
>---
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/haskell/cabal/pull/3469

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

File: travis-script.sh [hunk 1]
```diff
@@ -49,95 +49,85 @@ case $GHCVER in
...
+    echo "----"
+}
+
+# ---------------------------------------------------------------------
 # Bootstrap cabal, to verify bootstrap.sh script works.
 # ---------------------------------------------------------------------
 
-OLD_CWD=$PWD
+bootstrap() {
+    OLD_CWD=$PWD
```
Comment (BardurArantsson) [inline travis-script.sh]: Might it be a better idea to use subshells to ensure the preservation of CWD?

(I mean, the build will certainly fail if it isn't restored properly, but...)

**Comment by BardurArantsson** [comment]
I think it's a matter of pressing the "Merge" button? (One very minor comment, though. Not something that necessarily needs to changed)

**Comment by BardurArantsson** [comment]
Apologies if I overstepped any boundaries. LGTM, etc. and I think we should be actively investing into getting more self-insight (as it were).

**Pull request merged**